crates/milli/src/update/new/indexer/write.rs

```diff
@@ -218,6 +218,13 @@ pub fn write_from_bbqueue(
                     arroy_writers.get(&embedder_id).expect("requested a missing embedder");
                 let mut embeddings = Embeddings::new(*dimensions);
                 let all_embeddings = asvs.read_all_embeddings_into_vec(frame, aligned_embedding);
+                // FIXME: /!\ Case where number of embeddings is divisor of `dimensions` would still pass
+                if *dimensions!= 0 && all_embeddings.len() % *dimensions != 0 {
+                    return Err(Error::UserError(UserError::InvalidVectorDimensions {
+                        expected: *dimensions,
+                        found: all_embeddings.len(),
+                    }));
+                }
                 embeddings.append(all_embeddings.to_vec()).unwrap();
```

Comment:
```suggestion
                if embeddings.append(all_embeddings.to_vec()).is_err() {
                    return Err(Error::UserError(UserError::InvalidVectorDimensions {
                        expected: *dimensions,
                        found: all_embeddings.len(),
                    }));
                }
```

Instead of rewriting the code inside of `embeddings.append` I would prefer that you just use the result it returns.